packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin.tsx

```diff
@@ -47,6 +47,22 @@ export const TooltipPlugin: React.FC<TooltipPluginProps> = ({
     pluginLog(pluginId, true, `Focused series: ${focusedSeriesIdx}, focused point: ${focusedPointIdx}`);
   }, [focusedPointIdx, focusedSeriesIdx]);
 
+  useEffect(() => {
+    const plotMouseLeave = () => {
+      setCoords(null);
+    };
+
+    if (plotCtx && plotCtx.plot) {
+      plotCtx.plot.root.querySelector('.u-over')!.addEventListener('mouseleave', plotMouseLeave);
+    }
+
+    return () => {
+      if (plotCtx && plotCtx.plot) {
+        plotCtx.plot.root.querySelector('.u-over')!.removeEventListener('mouseleave', plotMouseLeave);
+      }
+    };
```

Comment:
Feel more comfortable now with the event listeners than with onmouseleave, good stuff!